File: swift/03p1.swift
import Foundation

var claims: [String] = [];
while let claim = readLine() { claims.append(claim) }

var patches = Array(repeating: Array(repeating: 0, count: 1000), count: 1000)
for claim in claims {
  let claimEls = claim.components(separatedBy: " @ ")
  let claimSpec = claimEls[1]
  let specGeoEls = claimSpec.components(separatedBy: ": ")
  let startCoord = specGeoEls[0]
  let area = specGeoEls[1]
  let startCoordEls = startCoord.components(separatedBy: ",")
  let startX = Int(startCoordEls[0]) ?? 0
  let startY = Int(startCoordEls[1]) ?? 0
  let areaEls = area.components(separatedBy: "x")
  let endX = startX + (Int(areaEls[0]) ?? 0)
  let endY = startY + (Int(areaEls[1]) ?? 0)

  for y in startY..<endY {
    for x in startX..<endX { patches[y][x] += 1 }
  }
}

var multipleClaimCount = 0
for row in patches {
  for c in row {
    if c > 1 { multipleClaimCount += 1 }
  }
}
print("\(multipleClaimCount)")

File: python/03p1.py
#!/usr/bin/env python

import sys

claims = [claim.rstrip() for claim in sys.stdin]
SQUARE_WIDTH = 1000
patches = [[0 for x in range(SQUARE_WIDTH)] for y in range(SQUARE_WIDTH)]
for claim in claims:
    claim_els = claim.split(" @ ")
    claim_spec = claim_els[1]
    spec_geo_els = claim_spec.split(": ")
    start_coord = spec_geo_els[0]
    area = spec_geo_els[1]
    start_coord_els = start_coord.split(",")
    start_x = int(start_coord_els[0]) or 0
    start_y = int(start_coord_els[1]) or 0
    area_els = area.split("x")
    end_x = int(start_x) + (int(area_els[0]) or 0)
    end_y = int(start_y) + (int(area_els[1]) or 0)

    for y in range(start_y, end_y):
        for x in range(start_x, end_x):
            patches[y][x] += 1

multiple_claim_count = 0
for cols in patches:
    for c in cols:
        if c > 1: multiple_claim_count += 1

print(multiple_claim_count)
